src/test/java/com/nccgroup/loggerplusplus/exports/CobaltExporterTest.java

```diff
@@ -0,0 +1,66 @@
+
+import com.coreyd97.BurpExtenderUtilities.Preferences;
+import com.nccgroup.loggerplusplus.logentry.LogEntry;
+import com.nccgroup.loggerplusplus.logentry.LogEntryField;
+import com.nccgroup.loggerplusplus.logentry.Status;
+import com.nccgroup.loggerplusplus.util.Globals;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.entity.StringEntity;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.management.*", "javax.swing.*", "javax.net.ssl.*"})
+@PrepareForTest(CobaltExporter.class)
+public class CobaltExporterTest {
+    LogEntry logEntry = mock(LogEntry.class);
+    Preferences preferences = mock(Preferences.class);
+    ExportController exportController = mock(ExportController.class);
+    CobaltExporter cobaltExporter;
+
+    @Before
+    public void setUp() {
+        List<LogEntryField> fields = new ArrayList<LogEntryField>();
+        fields.add(LogEntryField.QUERY);
+        when(preferences.getSetting(Globals.PREF_PREVIOUS_COBALT_FIELDS)).thenReturn(fields);
+        when(preferences.getSetting(Globals.PREF_COBALT_AUTOSTART_GLOBAL)).thenReturn(true);
+        when(preferences.getSetting(Globals.PREF_COBALT_DELAY)).thenReturn(1);
+        when(preferences.getSetting(Globals.PREF_COBALT_ADDRESS)).thenReturn("http://example.com");
+        cobaltExporter = new CobaltExporter(exportController, preferences);
+        when(logEntry.getStatus()).thenReturn(Status.PROCESSED);
+    }
+
+    @Test
+    public void testbuildRequestBody() throws Exception {
+        when(logEntry.getValueByKey(LogEntryField.QUERY)).thenReturn("a=1&b=1");
+        List entries = new ArrayList<LogEntry>();
+        entries.add(logEntry);
+        StringEntity result = Whitebox.invokeMethod(cobaltExporter, "buildRequestBody", entries);
+        String content = IOUtils.toString(result.getContent(), StandardCharsets.UTF_8);
+        assertEquals("[{\"Request.Query\":\"a=1&b=1\"}]", content);
+    }
+
+    @Test
+    public void testDoesNotThrowExceptionForNull() throws Exception {
+        when(logEntry.getValueByKey(LogEntryField.QUERY)).thenReturn(null);
+        List entries = new ArrayList<LogEntry>();
+        entries.add(logEntry);
+        StringEntity result = Whitebox.invokeMethod(cobaltExporter, "buildRequestBody", entries);
+        String content = IOUtils.toString(result.getContent(), StandardCharsets.UTF_8);
+        assertEquals("[{}]", content);
```

Comment:
I was expecting it to be something like `"[{\"Request.Query\":null}]"`. I imagine that if there is a key and it appears in the cycle, then it might be important. But for most fields it probably means that they are not set. So it's just a remark, not a request from my part.

Reply:
Good question. 
For example in the below case, 
`QUERY` is not used in the http request so its null. I think storing this field with null value in sumologic will complicate things when we start to process those logs and also it will remain an extra data that we need to ignore :) 
<img width="1243" alt="Screenshot 2021-07-06 at 16 22 58" src="https://user-images.githubusercontent.com/22538802/125443213-dc8ad8d4-b550-44c5-868a-8e7974ab843b.png">